### ISSUE TITLE
Incorrect use of "epoch" in the Optimizing Model Parameters tutorial

### ISSUE THREAD

**Issue by chrsigg** [pytorch/tutorials#2126]
From the first paragraph of the [Optimizing Model Parameters](https://github.com/pytorch/tutorials/blob/master/beginner_source/basics/optimization_tutorial.py) tutorial:

> in each iteration (called an epoch) the model makes a guess about the output, calculates the error in its guess (loss), collects the derivatives of the error with respect to its parameters (as we saw in the [previous section](https://pytorch.org/tutorials/beginner/basics/autograd_tutorial.html)), and optimizes these parameters using gradient descent.

What is described in this paragraph is a single optimization step. An epoch is a full pass over the dataset (see e.g. https://deepai.org/machine-learning-glossary-and-terms/epoch).

I propose to simply remove the "(called an epoch)" here, as the term is correctly used and explained later in the "Hyperparameters" section:

> Number of Epochs - the number times to iterate over the dataset

cc @suraj813

**Comment by suraj813** [comment]
Good catch! You are correct, that "epoch" was being used incorrectly in that statement. Thank you for bringing this up and helping to improve the quality of our tutorials :) If you see any other mistake, please raise an issue. If you know the fix, I'd encourage you to submit a PR and I'll be happy to merge your changes in.
